app/controllers/pages/additional_guidance_controller.rb

```diff
@@ -47,6 +47,13 @@ def update
     end
   end
 
+  def render_preview
+    markdown = params[:guidance_markdown]
+    preview_html = markdown.present? ? GovukFormsMarkdown.render(markdown) : "<p>You have no content to preview</p>"
```

Comment:
Could you not maybe create a new instance the guidance_form which just sets `guidance_markdown` and then reuse `preview_html` method?

Reply:
@aliuk2012 Good suggestion, have updated!